### ISSUE TITLE
Python mode doesn't recognize `==, >=, <=, !=, +=, *=` as operators.

### ISSUE THREAD

**Issue by ruoyu0088** [codemirror/CodeMirror#4876]
Here is the `doubleOperators` define in `python.js`:

```
var doubleOperators = parserConf.doubleOperators || /^([!<>]==|<>|<<|>>|\/\/|\*\*)/;
```

There is a bug in it for `!=, <=, >=, ==` operators, it should be:

```
var doubleOperators = parserConf.doubleOperators || /^([!<>=]=|<>|<<|>>|\/\/|\*\*)/;
```

and I thinks operators in `doubleDelimiters` should be appended to `doubleOperators`, because `+=` is also an operator in Python.

Here is the same issue in Jupyter notebook.

https://github.com/jupyter/notebook/issues/2682

**Comment by marijnh** [comment]
Attached patch should help with this. Does it solve your issue?

**Comment by adrianheine** [comment]
@marijnh  I added tests and commented on your change. Do you actually like having tests for such things?

**Comment by marijnh** [comment]
Yes, mode tests are great to have -- I'm often too lazy to add them, when I'm not too worried about regressions, but it's not that I don't like them.
